lib/src/importer/canonicalize_context.dart

```diff
@@ -0,0 +1,24 @@
+// Copyright 2014 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:meta/meta.dart';
+
+class CanonicalizeContext {
```

Comment:
Please document this and its members. Also, it should be `final class` and marked as `@internal`.